soapui/src/main/java/com/eviware/soapui/impl/actions/RestServiceBuilder.java

```diff
@@ -64,27 +67,79 @@ public String getUri() {
+        RestRequest restRequest = addNewRequest(addNewMethod(ModelCreationStrategy.CREATE_NEW_MODEL,
+                restResource, RestRequestInterface.HttpMethod.GET));
         copyParameters(extractParams(URI), restResource.getParams());
-        UISupport.select(restRequest);
-        UISupport.showDesktopPanel(restRequest);
-
+        if (showRequest) {
+            UISupport.select(restRequest);
+            UISupport.showDesktopPanel(restRequest);
+        }
+        return restRequest;
     }
 
-    public RestRequest createRestServiceHeadlessFromUri(WsdlProject project, RequestInfo requestInfo, ModelCreationStrategy methodReuseStrategy) throws MalformedURLException {
+    public RestRequest createRestServiceHeadlessFromUri(WsdlProject project, RequestInfo requestInfo,
+                                                        ModelCreationStrategy methodReuseStrategy) throws MalformedURLException {
         RestResource restResource = createResource(REUSE_MODEL, project, requestInfo.getUri());
         RestMethod restMethod = addNewMethod(methodReuseStrategy, restResource, requestInfo.getRequestMethod());
         RestRequest restRequest = addNewRequest(restMethod);
         copyParametersWithDefaultsOnResource(extractParams(requestInfo.getUri()), restMethod.getParams(), restRequest.getParams());
         return restRequest;
     }
 
+    public RestRequest createRestServiceWithMethod(WsdlProject project, String URI,
+                                                   RestRequestInterface.HttpMethod method,
+                                                   boolean showDesktopPanel,
+                                                   String requestName) throws MalformedURLException {
+        if (StringUtils.isNullOrEmpty(URI)) {
+            throw new MalformedURLException("The URL is null or empty");
+        }
+        RestResource restResource = createResource(REUSE_MODEL, project, URI);
+        ModelCreationStrategy methodReuseStrategy = CREATE_NEW_MODEL;
+        for (RestMethod restMethod : restResource.getRestMethodList()) {
+            if (restMethod.getMethod() == method) {
+                methodReuseStrategy = REUSE_MODEL;
+            }
+        }
+        RestMethod restMethod = addNewMethod(methodReuseStrategy, restResource, method);
+        RestRequest restRequest;
+        if (requestName != null) {
+            restRequest = restMethod.addNewRequest(requestName);
+        } else {
+            restRequest = addNewRequest(restMethod);
+        }
+        copyParameters(extractParams(URI), restMethod.getParams());
+        if (showDesktopPanel) {
+            UISupport.select(restRequest);
+            UISupport.showDesktopPanel(restRequest);
+        }
+        return restRequest;
+    }
+
+    public RestRequest createRestServiceFromInspectionData(WsdlProject project, String URI,
```

Comment:
do you call this method externally? should it be public?

Reply:
yes, in SaveRequestAction class